### ISSUE TITLE
Enable Offline Access to Unmigrated Accounts

### ISSUE THREAD

**Issue by ghost** [PolyMC/PolyMC#427]
### Role

I'm a Minecraft player with an unmigrated account.

### Suggestion

Enable the ability to use offline mode if the user's account is valid but unmigrated.

### Benefit

To keep some semblance of access to the game I paid for.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

Users have lost access to their ability to play due to the Microsoft acquisition and subsequent modification of the login system, there's nothing PolyMC can do about that. 

However, I propose enabling the offline account system if the user's account reports "Minecraft Java profile acquisition failed." on login. This is a different error code than an invalid username/password combination and can be used to infer a legitimate Minecraft purchase.

https://github.com/PolyMC/PolyMC/issues/422

**Comment by forkiesassds** [comment]
Isn't this considered piracy?

**Comment by ghost** [comment]
> Isn't this considered piracy?

No, the whole point of this system is to verify a legitimate Minecraft purchase.

**Comment by Un1q32** [comment]
why not just migrate your account?

**Comment by fn2006** [comment]
I'm in the same boat as you, I ended up just creating a fork that removes the DRM from the offline mode.

**Comment by Scrumplex** [comment]
Please don't promote projects that violate Minecraft's EULA.

**Comment by TheMCNerd2017** [comment]
> why not just migrate your account?

Some people can't or won't migrate for various reasons including:

1) They don't want to give more personal information to Microsoft.
2) They live in a country where Xbox Services is blocked for Minecraft or blocked in general.
3) They don't have access to or don't want to provide an SMS-capable phone number.
4) They don't agree to the Microsoft/Xbox Live TOS which bans anonymous alternate accounts along with account sharing and is in general far more strict than the old Mojang/Minecraft TOS.

**Comment by forkiesassds** [comment]
> > why not just migrate your account?
> 
> Some people can't or won't migrate for various reasons including:
> 
>     1. They don't want to give more personal information to Microsoft.
> 
>     2. They live in a country where Xbox Services is blocked for Minecraft or blocked in general.
> 
>     3. They don't have access to or don't want to provide an SMS-capable phone number.
> 
>     4. They don't agree to the Microsoft/Xbox Live TOS which bans anonymous alternate accounts along with account sharing and is in general far more strict than the old Mojang/Minecraft TOS.

5. They lost access to their email, forgot their security questions and/or lost their transaction ID.

**Comment by LapisDemon** [comment]
> No, the whole point of this system is to verify a legitimate Minecraft purchase.

The majority of my 20ish MCJE accs are by now blocked on API level, and I'm not sure if it would be really possible to really verify an actual purchase.. I'm not so deep into the technical aspects of it, but I'd like to point out that it's certain - with 100% - that MS will move all servers from AWS to Azure asap, now after the forcedMigration period started, and from then on out their next steps will unfold, which could become dangerous for MCJE at some point, if their future goals will be as I personally can fathom them.

Because at some point any external launchers might cease to exist altogether (I'll explain that somewhen), which is one of several reasons why I told my followers who are technically capable to, to download _all_ MCJE resources (means, incl. also all libraries etc.) asap.

> However, I propose enabling the offline account system if the user's account reports "Minecraft Java profile acquisition failed." on login. This is a different error code than an invalid username/password combination and can be used to infer a legitimate Minecraft purchase.

Keeping in mind what I just wrote, the question is:
For how long?

Of course, it'd be a temporary relief as locked but legit MCJE user to be able to play MCJE again (in case you can't figure out how to remove that specific part in the launcher yourself), but it would just be a short-term remedy and wouldn't deal with the actual issue at hand, just postpone it.

**Comment by dschemp** [comment]
If I may give my opinion on this: this should not be implemented.
If this is not a legal self-nuke than it is at least a legal grey zone. Nobody wants to risk going this far for some people and imo rightfully so. If you want this feature so desperately you will need to implement it on your own.
Other than that: afaik you are limited to the options of playing Minecraft singleplayer or a small amount of multiplayer servers (I am gonna be honest, some time has passed since I actively played MC) and that's it. And sooner or later your proposal, checking the error message and going into offline mode, won't work anymore. Mojang / Microsoft is at some point going to pull the plug on the Mojang Auth services and what are you gonna do then?

**Comment by LapisDemon** [comment]
> If this is not a legal self-nuke than it is at least a legal grey zone. Nobody wants to risk going this far for some people and imo rightfully so.

100%. I did upvote this issue as I feel for affected players who can't do it on their own, but I don't want to see Poly going into legal issues, potentially, as it's currently one of few ways how to not send telemetry for Migrated (aside from also e.g. Optifine who recently inserted that option to toggle off).

> Mojang / Microsoft is at some point going to pull the plug on the Mojang Auth services and what are you gonna do then?

Exactly what I tried to express with my prior comment.

So even if Poly would be willing to do this, it's not enough to postpone this issue, the predicament MCJE is in.

We've got to try to improve on the general situation itself MCJE is in.

**Comment by ZekeZDev** [comment]
I have added the wonfix label as I dont see this as anything we will implement but I think this discussion is valuable.

**Comment by TheCodex6824** [comment]
From my experimenting, I am able to get most of the Mojang account functionality back by changing the login flow to not use the profile endpoint. Yggdrasil (the actual authentication part) works fine, the only locked parts are the Minecraft-specific endpoints. To be more specific:

- Launching the game works
- Multiplayer (with normal Mojang authentication, so just as usual) works
- Changing skins does ***not*** work, at least from what I have tried
- Capes are probably in the same boat as skins but I don't have any so I couldn't try

I can open a PR if you want, but I figured I should probably ask first given this might make certain rude game companies not happy.

Edit: PR opened (#482)

**Comment by DavidDGTNT** [comment]
> > > why not just migrate your account?
> > 
> > Some people can't or won't migrate for various reasons including:
> > 
> >     1. They don't want to give more personal information to Microsoft.
> > 
> >     2. They live in a country where Xbox Services is blocked for Minecraft or blocked in general.
> > 
> >     3. They don't have access to or don't want to provide an SMS-capable phone number.
> > 
> >     4. They don't agree to the Microsoft/Xbox Live TOS which bans anonymous alternate accounts along with account sharing and is in general far more strict than the old Mojang/Minecraft TOS.
> 
> 5. They lost access to their email, forgot their security questions and/or lost their transaction ID.

I had the first issue, but fixed it by contacting support. I didn't have the second.
